### PR TITLE
(CDAP-3932) Added app.program.extra.classpath option

### PR DESCRIPTION
**Pull request by chtyim**
- The option carries extra classpath to be used for all programs
- The paths specified in the classpath are included when
  in the program class loader.

### CODE REVIEW

File: cdap-common/src/main/java/co/cask/cdap/common/lang/DirectoryClassLoader.java [hunk 1]
```diff
@@ -92,6 +97,19 @@ public Manifest getManifest() {
       for (String libDir : libDirs) {
         addJarURLs(new File(dir, libDir), urls);
       }
+
+      if (extraClassPath != null) {
+        for (String classPath : Splitter.on(File.pathSeparatorChar).omitEmptyStrings().split(extraClassPath)) {
```
Comment (jwang47) [inline cdap-common/src/main/java/co/cask/cdap/common/lang/DirectoryClassLoader.java]: Maybe `classPathEntry` would be a better name here.

**Comment by jwang47** [comment]
One minor comment, otherwise LGTM.

File: cdap-common/src/main/java/co/cask/cdap/common/lang/DirectoryClassLoader.java (hunk 1 above)
Comment (chtyim) [inline cdap-common/src/main/java/co/cask/cdap/common/lang/DirectoryClassLoader.java]: Renamed to `path`.

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-RBT501-1

**Pull request merged**